### PR TITLE
[PDI-8927] Trans.activeSubtransformations not threadsafe

### PR DESCRIPTION
**Pull request by IvanNikolaychuk**
- Changing HM to CHM for synchronization
- Separating add/remove/get methods, encapsulating map-implementation
- Test written
- Checkstyle applied

### CODE REVIEW

**Comment by IvanNikolaychuk** [comment]
@mchen-len-son , could you please review this PR?

**Comment by IvanNikolaychuk** [comment]
@mchen-len-son , @bmorrise, could you please review this PR?

**Comment by IvanNikolaychuk** [comment]
HashMap's import is not unused - it's used in java docs.

File: engine/src/org/pentaho/di/trans/Trans.java [hunk 1]
```diff
@@ -5136,14 +5136,31 @@ public void setLogLevel( LogLevel logLevel ) {
   }
 
   /**
-   * Gets the active sub-transformations.
+   *  Use:
+   *  {@link #addActiveSubTransformation(String, Trans),
+   *  {@link #getActiveSubTransformation(String)},
+   *  {@link #removeActiveSubTransformation(String)}
+   *  {@link #getNumberOfActiveSubTransformations()}
```
Comment (mchen-len-son) [inline engine/src/org/pentaho/di/trans/Trans.java]: no such method

Comment (IvanNikolaychuk) [inline engine/src/org/pentaho/di/trans/Trans.java]: oops, yeah

**Comment by mchen-len-son** [comment]
Looks like wingman has been running on "bad" changes too long, and therefore reported after the "good" changes.
Merging.

**Pull request merged**